lib/shopify_app/admin_api/with_token_refetch.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module ShopifyApp
+  module AdminAPI
+    module WithTokenRefetch
+      def with_token_refetch(session, session_token)
+        retrying = false if retrying.nil?
+        yield
+      rescue ShopifyAPI::Errors::HttpResponseError => error
+        if error.code == 401 && !retrying
+          retrying = true
```

Comment:
Does it make more sense to move the delete session logic to here instead of the concern? If apps use this wrapper method and it failed, it the session should be deleted right?? 🤔 
```
ShopifyApp::SessionRepository.delete_session(session.id)
```